File: frontend/src/modules/labour/services/workerPaymentService.ts
import { apiGet, apiPost } from '@/services/apiClient';
import type { WorkerPaymentRequest, WorkerPaymentResponse, WorkerWageSummaryResponse } from '../types';

/** Backend: labour/controller/WorkerPaymentController.java */
export const workerPaymentService = {
  create: (body: WorkerPaymentRequest) => apiPost<WorkerPaymentResponse>('/v1/worker-payments', body),

  listForWorker: (workerId: number) =>
    apiGet<WorkerPaymentResponse[]>(`/v1/workers/${workerId}/payments`),

  /** Soft cancel - the payment stays in history but stops counting towards the paid total. */
  cancel: (id: number) => apiPost<void>(`/v1/worker-payments/${id}/cancel`),

  wageSummary: (workerId: number, fromDate?: string, toDate?: string) =>
    apiGet<WorkerWageSummaryResponse>(`/v1/workers/${workerId}/wage-summary`, { params: { fromDate, toDate } }),
};

File: backend/src/main/java/com/hardware/erp/labour/service/WorkerPaymentService.java
package com.hardware.erp.labour.service;

import com.hardware.erp.labour.dto.WorkerPaymentRequest;
import com.hardware.erp.labour.dto.WorkerPaymentResponse;
import com.hardware.erp.labour.dto.WorkerWageSummaryResponse;

import java.time.LocalDate;
import java.util.List;

public interface WorkerPaymentService {

    WorkerPaymentResponse create(WorkerPaymentRequest request);

    List<WorkerPaymentResponse> listForWorker(Long workerId);

    /** Soft cancel - the row stays in history, marked CANCELLED, and stops counting towards the worker's paid total. */
    void cancel(Long id);

    /** Wages earned (live-computed from attendance x current daily rate) vs paid, for one worker over a date range. */
    WorkerWageSummaryResponse wageSummary(Long workerId, LocalDate fromDate, LocalDate toDate);
}
